### PR TITLE
Upgrade SDK

### PR DESCRIPTION
**Pull request by ryanbrandenburg**

### CODE REVIEW

**Comment by natemcmaster** [comment]
Bummer, travis CI is not happy.

**Comment by ryanbrandenburg** [comment]
Yeah, this passes locally for me, and it seems like the kind of failure that would be all or nothing, so it's pretty strange.

**Comment by ryanbrandenburg** [comment]
So, this is apparently a timeout, we took longer than 5 minutes to complete just this test, whereas previously the entire build could complete in ~2-3 minutes.

**Comment by ryanbrandenburg** [comment]
I can replicate this by running build.cmd on an Ubuntu machine.

**Comment by natemcmaster** [comment]
That's not great

File: test/KoreBuild.FunctionalTests/Utilities/TestApp.cs [hunk 1]
```diff
@@ -41,7 +41,7 @@ public TestApp(string templateDir, string toolsSource, string source, string wor
             return await ExecuteScript(output, "build", new string[0], commandArgs);
         }
 
-        private async Task<int> ExecuteScript(ITestOutputHelper output, string script, string[] koreBuildArgs, params string[] commandArgs)
+        private Task<int> ExecuteScript(ITestOutputHelper output, string script, string[] koreBuildArgs, params string[] commandArgs)
```
Comment (natemcmaster) [inline test/KoreBuild.FunctionalTests/Utilities/TestApp.cs]: This method is no longer asynchronous. WaitForExit is blocking.

Comment (natemcmaster) [reply test/KoreBuild.FunctionalTests/Utilities/TestApp.cs]: I recommend just making the return type `int` instead of `Task<int>`

Comment (ryanbrandenburg) [reply test/KoreBuild.FunctionalTests/Utilities/TestApp.cs]: I know, I was using the branch to test out a change on my Ubuntu VM, I'll :up::date: when I'm proposing a fix.

**Comment by ryanbrandenburg** [comment]
@natemcmaster and @jbagga for approval.

**Comment by jbagga** [comment]
I have been trying this SDK version and it leads to test failures in templating
```
System.Exception : Process exited with code 1
StdErr: 
StdOut: System.TypeLoadException: Method 'FindMapping' in type 'Microsoft.EntityFrameworkCore.Sqlite.Storage.Internal.SqliteTypeMappingSource' from assembly 'Microsoft.EntityFrameworkCore.Sqlite, Version=2.1.0.0, Culture=neutral, PublicKeyToken=adb9793829ddae60' does not have an implementation.
   at Microsoft.Extensions.DependencyInjection.SqliteServiceCollectionExtensions.AddEntityFrameworkSqlite(IServiceCollection serviceCollection)
   at Microsoft.EntityFrameworkCore.Sqlite.Infrastructure.Internal.SqliteOptionsExtension.ApplyServices(IServiceCollection services)
   at Microsoft.EntityFrameworkCore.Internal.ServiceProviderCache.ApplyServices(IDbContextOptions options, ServiceCollection services)
   at Microsoft.EntityFrameworkCore.Internal.ServiceProviderCache.<>c__DisplayClass4_0.<GetOrAdd>b__2(Int64 k)
   at System.Collections.Concurrent.ConcurrentDictionary`2.GetOrAdd(TKey key, Func`2 valueFactory)
   at Microsoft.EntityFrameworkCore.Internal.ServiceProviderCache.GetOrAdd(IDbContextOptions options, Boolean providerRequired)
   at Microsoft.EntityFrameworkCore.DbContext..ctor(DbContextOptions options)
   at Microsoft.AspNetCore.Identity.EntityFrameworkCore.IdentityDbContext..ctor(DbContextOptions options)
   at AspNet.Template.fbedacc119b042b4a1e5b1ae24d383b5.Data.ApplicationDbContext..ctor(DbContextOptions`1 options) in C:\Test\templating\test\Templates.Test\bin\Debug\netcoreapp2.1\TestTemplates\AspNet.Template.fbedacc119b042b4a1e5b1ae24d383b5\Data\ApplicationDbContext.cs:line 12
--- End of stack trace from previous location where exception was thrown ---
   at Microsoft.Extensions.DependencyInjection.ServiceLookup.CallSiteRuntimeResolver.VisitConstructor(ConstructorCallSite constructorCallSite, ServiceProviderEngineScope scope)
   at Microsoft.Extensions.DependencyInjection.ServiceLookup.CallSiteVisitor`2.VisitCallSite(IServiceCallSite callSite, TArgument argument)
   at Microsoft.Extensions.DependencyInjection.ServiceLookup.CallSiteRuntimeResolver.VisitScoped(ScopedCallSite scopedCallSite, ServiceProviderEngineScope scope)
   at Microsoft.Extensions.DependencyInjection.ServiceLookup.CallSiteVisitor`2.VisitCallSite(IServiceCallSite callSite, TArgument argument)
   at Microsoft.Extensions.DependencyInjection.ServiceLookup.DynamicServiceProviderEngine.<>c__DisplayClass1_0.<RealizeService>b__0(ServiceProviderEngineScope scope)
   at Microsoft.Extensions.DependencyInjection.ServiceLookup.ServiceProviderEngine.GetService(Type serviceType, ServiceProviderEngineScope serviceProviderEngineScope)
   at Microsoft.Extensions.DependencyInjection.ServiceLookup.ServiceProviderEngineScope.GetService(Type serviceType)
   at Microsoft.Extensions.DependencyInjection.ActivatorUtilities.GetServiceOrCreateInstance(IServiceProvider provider, Type type)
   at Microsoft.EntityFrameworkCore.Design.Internal.DbContextOperations.<>c__DisplayClass12_2.<FindContextTypes>b__11()
   at Microsoft.EntityFrameworkCore.Design.Internal.DbContextOperations.CreateContext(Func`1 factory)
   at Microsoft.EntityFrameworkCore.Design.Internal.DbContextOperations.CreateContext(String contextType)
   at Microsoft.EntityFrameworkCore.Design.Internal.MigrationsOperations.AddMigration(String name, String outputDir, String contextType)
   at Microsoft.EntityFrameworkCore.Design.OperationExecutor.AddMigrationImpl(String name, String outputDir, String contextType)
   at Microsoft.EntityFrameworkCore.Design.OperationExecutor.AddMigration.<>c__DisplayClass0_1.<.ctor>b__0()
   at Microsoft.EntityFrameworkCore.Design.OperationExecutor.OperationBase.<>c__DisplayClass3_0`1.<Execute>b__0()
   at Microsoft.EntityFrameworkCore.Design.OperationExecutor.OperationBase.Execute(Action action)
Method 'FindMapping' in type 'Microsoft.EntityFrameworkCore.Sqlite.Storage.Internal.SqliteTypeMappingSource' from assembly 'Microsoft.EntityFrameworkCore.Sqlite, Version=2.1.0.0, Culture=neutral, PublicKeyToken=adb9793829ddae60' does not have an implementation.

```

cc @smitpatel @bricelam

**Comment by natemcmaster** [comment]
> I have been trying this SDK version and it leads to test failures in templating

Isn't templating supposed to be decoupled from the SDK version now? I thought we made that change.

**Comment by smitpatel** [comment]
@jbagga That method still exists in `SqliteTypeMappingSource`. Though there is difference between 2.1 & dev that the method in dev takes readonly struct. Is there issue with package version mismatch?

**Comment by jbagga** [comment]
We made a change for dotnet-ef package. Not other EF dependencies. Yes looks like a version mismatch. I'll investigate. Thanks

**Pull request merged**